share/availability/full/options.go

```diff
@@ -0,0 +1,41 @@
+package full
+
+import (
+	"time"
+
+	"github.com/celestiaorg/celestia-node/share/availability"
+)
+
+type params struct {
+	storageWindow time.Duration
+	archival      bool
+}
+
+// Option is a function that configures light availability Parameters
+type Option func(*params)
+
+// DefaultParameters returns the default Parameters' configuration values
+// for the light availability implementation
+func defaultParams() *params {
+	return &params{
+		storageWindow: availability.StorageWindow,
+		archival:      false,
+	}
+}
+
+// WithArchivalMode is a functional option to tell the full availability
+// implementation that the node wants to sync *all* blocks, not just those
+// within the sampling window.
+func WithArchivalMode() Option {
+	return func(p *params) {
+		p.archival = true
+	}
+}
+
+// WithStorageWindow is a functional option to set the storage window
+// to something other than the default for the full availability implementation
+func WithStorageWindow(window time.Duration) Option {
+	return func(p *params) {
+		p.storageWindow = window
+	}
+}
```

Comment:
This one looks like only be used in tests. And I think it should not be exposed to user, since we rely on its value on shrex clients. How about remove it from params and put on availability struct with hardcoded values (similar to light availability)?

Reply:
Yeah i was thinking to do that too - will do.